portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -314,6 +314,37 @@ async function loadSearch() {
   insertSearch(otherElements);
 }
 
+/**
+ * Gets links from the navigation bar
+ * Removes link for current page we are on
+ * @return {array}: the links from other pages.
+ */
+function getOtherLinks() {
+  const pagePath = window.location.pathname;
+  const pathSize = pagePath.length;
+
+  let linkElements =
+      document.getElementById('navBar').getElementsByTagName('a');
+  linkElements = Array.from(linkElements);
+  let numLinks = linkElements.length;
+
+  for (let i = 0; i < numLinks; i++) {
+    const link = linkElements[i].href;
+    if (pathSize < 2) {
+      if (link.includes('index')) {
+        linkElements.splice(i, 1);
+        numLinks--;
```

Comment:
This code is duplicated in two places, so consider whether you can structure your code so that they're in the same if clause.

Reply:
I won't be able to have them in the same if clause because there is a non-generic scenario when we load the index page for the first time and the page path is '\' Once you navigate to the index page again the page path will be '\index.html'